src/RecyclableMemoryStreamManager.cs

```diff
@@ -801,51 +786,56 @@ public MemoryStream GetStream(string tag, Memory<byte> buffer)
         /// <summary>
         /// Triggered when a new block is created.
         /// </summary>
-        public event EventHandler BlockCreated;
-
-        /// <summary>
-        /// Triggered when a new block is created.
-        /// </summary>
-        public event EventHandler BlockDiscarded;
+        public event Action<BlockCreatedEventArgs> BlockCreated;
 
         /// <summary>
         /// Triggered when a new large buffer is created.
         /// </summary>
-        public event EventHandler LargeBufferCreated;
+        public event Action<LargeBufferCreatedEventArgs> LargeBufferCreated;
 
         /// <summary>
         /// Triggered when a new stream is created.
         /// </summary>
-        public event EventHandler StreamCreated;
+        public event Action<StreamCreatedEventArgs> StreamCreated;
 
         /// <summary>
         /// Triggered when a stream is disposed.
         /// </summary>
-        public event EventHandler StreamDisposed;
+        public event Action<StreamDisposedEventArgs> StreamDisposed;
+
+        /// <summary>
+        /// Triggered when a stream is disposed of twice (an error).
+        /// </summary>
+        public event Action<StreamDoubleDisposedEventArgs> StreamDoubleDisposed;
 
         /// <summary>
         /// Triggered when a stream is finalized.
         /// </summary>
-        public event EventHandler StreamFinalized;
+        public event Action<StreamFinalizedEventArgs> StreamFinalized;
 
         /// <summary>
         /// Triggered when a stream is finalized.
         /// </summary>
-        public event StreamLengthReportHandler StreamLength;
+        public event Action<StreamLengthEventArgs> StreamLength;
 
         /// <summary>
         /// Triggered when a user converts a stream to array.
         /// </summary>
-        public event EventHandler StreamConvertedToArray;
+        public event Action<StreamConvertedToArrayEventArgs> StreamConvertedToArray;
+
+        /// <summary>
+        /// Triggered when a stream is requested to expand beyond the maximum length specified by the responsible RecyclableMemoryStreamManager.
+        /// </summary>
+        public event Action<StreamOverCapacityEventArgs> StreamOverCapacity;
 
         /// <summary>
-        /// Triggered when a large buffer is discarded, along with the reason for the discard.
+        /// Triggered when a buffer of either type is discarded, along with the reason for the discard.
         /// </summary>
-        public event LargeBufferDiscardedEventHandler LargeBufferDiscarded;
+        public Action<BufferDiscardedEventArgs> BufferDiscarded;
```

Comment:
Why isn't this an event anymore?

Reply:
woops, oversight

Reply:
Any reason these aren't `EventHandler<TEventArgs>`?